include/irr/video/IGPUMeshBuffer.h

```diff
@@ -20,6 +20,14 @@ namespace video
 {
 	class IGPUMeshBuffer final : public asset::IMeshBuffer<IGPUBuffer,IGPUDescriptorSet,IGPURenderpassIndependentPipeline>
 	{
+    public:
+        _IRR_STATIC_INLINE_CONSTEXPR size_t PUSH_CONSTANTS_BYTESIZE = 128u;
+
+        uint8_t* getPushConstantsDataPtr() { return m_pushConstantsData; }
+        const uint8_t* getPushConstantsDataPtr() const { return m_pushConstantsData; }
+
+    private:
+        uint8_t m_pushConstantsData[PUSH_CONSTANTS_BYTESIZE];
```

Comment:
alignas to AVX512

Reply:
why the push constants only in IGPUMeshBuffer and not ICPUMeshBuffer ?

Reply:
because they're dynamic thing (most likely changes very often) so i think exporting them (as asset to file) is rather pointless

Reply:
there's probably no harm in serializing them... plus actually their use can be very very static (non-dynamic)

Because of a desire to re-use command buffers, and once recorded PCs can't change (have to record another cmd buffer), so I'd actually see them being used for super high-frequency (less overhead than dynamic UBO offset) but small and static (for the mesh) data, such as ObjectIDs, material IDs, colors, booleans, and other switches.

Reply:
resolved